cmd/server/app/migrate_up.go

```diff
@@ -127,10 +130,64 @@ var upCmd = &cobra.Command{
 			return fmt.Errorf("error preparing authz client: %w", err)
 		}
 
-		return nil
+		store := db.NewStore(dbConn)
+		return migrateProjectsRootAndMetadata(ctx, cmd, store)
 	},
 }
 
+func migrateProjectsRootAndMetadata(ctx context.Context, cmd *cobra.Command, store db.Store) error {
+	cmd.Println("Migrating projects to have no parent...")
+	projs, err := store.ListNonOrgProjects(ctx)
+	if err != nil {
+		return fmt.Errorf("error while listing non-org projects: %w", err)
+	}
+
+	for _, p := range projs {
+		m := projects.NewSelfEnrolledMetadata()
+		if err := json.Unmarshal(p.Metadata, &m); err != nil {
+			return fmt.Errorf("error while marshalling old metadata: %w", err)
+		}
+
+		newmeta, err := json.Marshal(&m)
+		if err != nil {
+			return fmt.Errorf("error while marshalling new metadata: %w", err)
+		}
+
+		if p.ParentID.Valid {
+			cmd.Printf("Migrating project %s to have root as parent\n", p.ID.String())
+			if _, err := store.OrphanProject(ctx, db.OrphanProjectParams{
+				ID:       p.ID,
+				Metadata: newmeta,
+			}); err != nil {
+				return fmt.Errorf("error while migrating project %s: %w", p.ID.String(), err)
+			}
+		} else {
+			cmd.Printf("Ensuring project has new metadata")
+			if _, err := store.UpdateProjectMeta(ctx, db.UpdateProjectMetaParams{
+				ID:       p.ID,
+				Metadata: newmeta,
+			}); err != nil {
+				return fmt.Errorf("error while updating project metadata: %w", err)
+			}
+		}
+	}
+
+	cmd.Println("Deleting old org projects...")
+	projs, err = store.ListOldOrgProjects(ctx)
+	if err != nil {
+		return fmt.Errorf("error while listing old org projects: %w", err)
+	}
+
+	for _, p := range projs {
+		_, err := store.DeleteProject(ctx, p.ID)
+		if err != nil {
+			return fmt.Errorf("error while deleting project: %w", err)
+		}
+	}
```

Comment:
I'm nervous about this with "ON DELETE CASCADE".  The previous _should_ have cleaned up all children, but I'd like to see something in the query or the code which will avoid deleting org-projects which *do* have children, particularly since all the above isn't in a transaction (so we could race and delete a newly-created project).